oscaralanpierce/skyrim_inventory_management_frontend
# Write generic user dashboard page

## Context

[**Write generic user dashboard page**](https://trello.com/c/4BcRZ4jO/244-write-generic-user-dashboard-page)

As we're moving on to the phase of the rewrite where we need to implement authentication, it will be helpful to have a page to put behind the authentication we implement. This page will be the dashboard main page. For the purposes of this PR, this page is undifferentiated - it doesn't contain any custom information for the user since we don't have a user yet.

Writing this has involved adding several new components as well as a new context, the `ColorContext`, and the corresponding hook, `useColorScheme`.

## Changes

* Create `DashboardHeader` component with stories and tests
* Create `NavigationCard` component with stories and tests
* Create `NavigationMosaic` component with stories and tests
* Create `ColorContext` and `useColorScheme` hook
* Create `DashboardLayout` layout component with stories and tests
* Create `DashboardPage` with stories and tests and wire up to page routes
* Create module for colour schemes
* Update PR template to include headings for all supported screen sizes under "Screenshots and GIFs"

## Considerations

I have mixed feelings about snapshot testing but decided to implement it for the purpose of these tests.

## Manual Test Cases

Run the site locally. Visit `http://localhost:5173/dashboard` and view at all supported viewport widths. Make sure the site looks like the screenshots.

## Screenshots and GIFs

### 320px

<img width="321" alt="Dashboard-320" src="https://user-images.githubusercontent.com/5115928/220454774-54ba372b-30e6-4617-90b6-6dcab4182113.png">

### 481px

<img width="481" alt="Dashboard-481" src="https://user-images.githubusercontent.com/5115928/220454812-e69e0ee5-d67d-496a-880a-744155db4241.png">

### 601px

<img width="601" alt="Dashboard-601" src="https://user-images.githubusercontent.com/5115928/220454857-2971eafd-0cfa-47d7-8025-e59ab915b049.png">

### 769px

<img width="769" alt="Dashboard-769" src="https://user-images.githubusercontent.com/5115928/220454920-3c2f868b-b213-43a9-a749-fb0b11c85605.png">

### 1025px

<img width="1025" alt="Dashboard-1025" src="https://user-images.githubusercontent.com/5115928/220454955-6e626af4-24ca-4ebb-82fe-5779131a665c.png">

### 1201px

<img width="1201" alt="Dashboard-1201" src="https://user-images.githubusercontent.com/5115928/220454989-b247d7af-3821-4e63-9a82-0a8a903a435d.png">

### 1405px

<img width="1405" alt="Dashboard-1405" src="https://user-images.githubusercontent.com/5115928/220455016-15b3b351-e857-4825-b4dd-11caa2516a3a.png">

### Large Desktop

<img width="1919" alt="Dashboard-Full" src="https://user-images.githubusercontent.com/5115928/220455054-cf45db04-ac94-4053-a083-30d7a66e94fa.png">
